### ISSUE TITLE
Protect against unexpected module changes

### ISSUE THREAD

**Issue by aahlenst** [adoptium/temurin-build#4390]
Our JDK 15 builds for Windows/x64 "lost" some modules along the way that were present in 14 and were supposed to be in 15. We didn't notice that until very late in the release cycle ([bug report](https://github.com/AdoptOpenJDK/openjdk-build/issues/2047)). We need test coverage that helps us detect such changes, especially if we introduce the JRE replacement that relies on curated module lists.

One possible way to achieve this: Store module lists per version, operating system and architecture and compare those lists with the output of `java --list-modules`. If there are differences, fail the test.

**Comment by smlambert** [comment]
A "Check number of modules" test seems like it can be part of our smoke tests in the temurin-build repository (where smoke tests check if we built the right thing, a prerequisite test, ahead of running the functional/openjdk/system/perf/jck tests).  Will transfer to temurin-build repo to see if someone wants to pick it up from there.

**Comment by karianna** [comment]
I don't think this has been an issue for a long, long time, I'm going to close this as not planned unless it rears its head again.
